clients/cobol-lsp-vscode-extension/src/services/CopybookURI.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as fs from "fs";
+import * as path from "path";
+import {URL} from "url";
+import * as vscode from "vscode";
+import {C4Z_FOLDER, COPYBOOKS_FOLDER, PATHS_LOCAL_KEY, PATHS_ZOWE, SETTINGS_SECTION} from "../constants";
+import {CopybooksDownloader} from "./CopybooksDownloader";
+import {ProfileService} from "./ProfileService";
+import {CopybookResolver} from "./settings/CopybookResolver";
+import {LocalCopybookResolver} from "./settings/LocalCopybookResolver";
+import {CopybookUtil} from "./settings/util/CopybookUtil";
+
+/**
+ * This class is responsible to identify from which source resolve copybooks required by the server.
+ */
+export class CopybookURI {
+    /**
+     * This method scan the list of folders as given input and find the required copybook name within the folder.
+     * If found returns its URI representation
+     * @param copybookName name of the copybook asked by the server
+     * @param targetFolders list of folders from where to search the copybook
+     */
+    public static searchInWorkspace(copybookName: string, targetFolders: string[]): string {
+        const copybookResolver: CopybookResolver = new LocalCopybookResolver();
```

Comment:
The LocalCopybookResolver doesn't have any internal state. Is it necessary to create an instance?
Could `resolve` be a static utility method?

Reply:
I will take a look at that

Reply:
The idea is to apply chain of responsability pattern there and leave to the appropriate derivatives the effort to resolve the copybook (local or endevor or zowe).
I will implement this enhancement later on.